lib/srv/desktop/discovery.go

```diff
@@ -168,18 +158,15 @@ func (s *WindowsService) getDesktopsFromLDAP() map[string]types.WindowsDesktop {
 		attrs = append(attrs, computerAttributes...)
 		attrs = append(attrs, discoveryConfig.LabelAttributes...)
 
-		entries, err := s.lc.ReadWithFilter(discoveryConfig.BaseDN, filter, attrs)
-		if trace.IsConnectionProblem(err) {
-			// If the connection was broken, re-initialize the LDAP client so that it's
-			// ready for the next reconcile loop. Return the last known set of desktops
-			// in this case, so that the reconciler doesn't delete the desktops it already
-			// knows about.
-			s.cfg.Logger.InfoContext(s.closeCtx, "LDAP connection error when searching for desktops, reinitializing client")
-			if err := s.initializeLDAP(); err != nil {
-				s.cfg.Logger.ErrorContext(s.closeCtx, "failed to reinitialize LDAP client, will retry on next reconcile", "error", err)
-			}
-			return s.lastDiscoveryResults
-		} else if err != nil {
+		tc, err := s.tlsConfigForLDAP()
+
+		lc := winpki.NewLDAPClient(s.cfg.LDAPConfig)
```

Comment:
Move this line after the error check below.

This way you keep the assignment of `err` right next to the code that reads it, and you can avoid the `NewLDAPClient` call completely if you got an error.